nbgrader/apps/dbapp.py

```diff
@@ -116,48 +116,106 @@ def start(self):
             gb.remove_student(student_id)
 
 
-class DbStudentImportApp(NbGrader):
+class DbGenericImportApp(NbGrader):
 
-    name = 'nbgrader-db-student-import'
-    description = 'Import students into the nbgrader database from a CSV file'
+    def get_db_update_method(self, gb):
+        raise NotImplementedError
 
-    aliases = aliases
-    flags = flags
+    @property
+    def name(self):
+        raise NotImplementedError
+
+    @property
+    def description(self):
+        raise NotImplementedError
+
+    @property
+    def table_class(self):
+        raise NotImplementedError
+
+    @property
+    def primary_key(self):
```

Comment:
Since this is only really being used for the student table, it seems like it would be easier just to set `primary_key = 'id'` (just like in the assignment class) rather than detecting it automatically.

Reply:
✓